### PR TITLE
Add write-tests skill

### PR DESCRIPTION
**Pull request by kirich1409**
## Summary

- New `write-tests` skill — retroactive test generation for existing code
- 6 phases: scope target → discover test infra → plan test cases → generate (kotlin-engineer) → verify → report
- Delegates code generation to kotlin-engineer, handles discovery and orchestration
- Consumes existing test plans from generate-test-plan (optional)

## Test plan

- [ ] Skill discoverable and auto-triggers on "write tests for this class"
- [ ] Does NOT trigger on "create a test plan" (that's generate-test-plan)
- [ ] Test infrastructure discovery works (detects JUnit/Kotest/MockK)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: plugins/developer-workflow/skills/write-tests/SKILL.md [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+- Public API surface (public/internal classes, functions, properties)
+- Dependencies (constructor parameters, injected services)
+- Complexity indicators (branching, state management, error handling)
+- Whether the code is Compose UI or non-UI Kotlin
+
+### 1.3 Find existing tests
+
+Search for existing tests:
+- Check the corresponding test source set (`src/test/`, `src/androidTest/`, `src/commonTest/`)
+- Use `ast-index search` or Grep to find test classes that reference the target classes
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/write-tests/SKILL.md]: Fixed in d9f42e6. Replaced Grep with ast-index for symbol search.

File: plugins/developer-workflow/skills/write-tests/SKILL.md [hunk 2]
```diff
@@ -0,0 +1,373 @@
...
+### 5.1 Run tests
+
+Run the test suite for the target module:
+
+```bash
+./gradlew :module:test
+```
+
+Or the more specific task if known (e.g., `:module:testDebugUnitTest`).
+
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/write-tests/SKILL.md:274]: Not addressing in this PR — the placeholder follows skill convention and the context makes the intent clear.

File: plugins/developer-workflow/README.md [hunk 3]
```diff
@@ -58,6 +58,20 @@ Guides a full migration of an Android module to Kotlin Multiplatform (KMP):
 
 Use when migrating a module to share code with iOS, JVM, or other platforms.
 
+### `write-tests`
+
+Writes retroactive tests for existing code that lacks coverage:
+- Accepts a file, class, module, or directory as target
+- Discovers the project's test infrastructure (framework, assertions, mocking, naming conventions)
+- Plans test cases for untested public API, edge cases, and error paths
+- Delegates code generation to `kotlin-engineer` (or `compose-developer` for Compose UI)
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/README.md:79]: Fixed in d9f42e6. Reworded to clarify the skill orchestrates test generation, not writes directly.

File: plugins/developer-workflow/skills/write-tests/SKILL.md [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/write-tests/SKILL.md:274]: Fixed in latest commit. Added connectedAndroidTest command for instrumentation tests alongside unit test command.

File: plugins/developer-workflow/skills/write-tests/SKILL.md [hunk 5]
```diff
@@ -0,0 +1,373 @@
...
+If any tests exposed real bugs in the target code, list them:
+```
+Findings:
+- LoginUseCase.login() does not check for empty password — allows login with blank credentials
+- AuthRepository.getUser() swallows IOException instead of propagating, returns stale cached data silently
+```
+
+Save findings to `swarm-report/<slug>-test-findings.md` only if production bugs were
+discovered. Format:
```
[the remark this answers is not in the document]
Comment (kirich1409) [reply plugins/developer-workflow/skills/write-tests/SKILL.md:350]: Fixed in latest commit. Added slug derivation step (1.1.1) with examples, consistent with other skills.
